core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java

```diff
@@ -683,6 +683,31 @@ private static boolean compareTimestampsAndDelete(
               CarbonTablePath.DataFileUtil.getTimeStampFromFileName(invalidFile.getName()));
     }
 
+    // This check is because, when there are some invalid files like tableStatusUpdate.write files
+    // present in store [[which can happen during delete or update if the disk is full or hdfs quota
+    // is finished]] then fileTimestamp will be null, in that case check for max query out and
+    // delete the .write file after timeout
+    if (fileTimestamp == null) {
+      String tableUpdateStatusFilename = invalidFile.getName();
+      if (tableUpdateStatusFilename.endsWith(".write")) {
+        long tableUpdateStatusFileTimeStamp = Long.parseLong(tableUpdateStatusFilename
+            .substring(tableUpdateStatusFilename.lastIndexOf(CarbonCommonConstants.HYPHEN) + 1,
+                invalidFile.getName().indexOf(".")));
+        if (isMaxQueryTimeoutExceeded(tableUpdateStatusFileTimeStamp)) {
+          try {
+            LOGGER.info("deleting the invalid .write file : " + invalidFile.getName());
```

Comment:
Please check if below method can be used because the code is same
org.apache.carbondata.core.mutate.CarbonUpdateUtil#compareTimestampsAndDelete

Reply:
done